client/my-sites/pages/page/index.js

```diff
@@ -424,51 +416,65 @@ const Page = React.createClass( {
 			</CompactCard>
 
 		);
-	},
+	}
 
-	updateStatusPublish: function() {
+	updateStatusPublish = () => {
 		this.props.updatePostStatus( 'publish' );
-		recordEvent( 'Clicked Publish Page' );
-	},
+		this.props.recordEvent( 'Clicked Publish Page' );
+	}
 
-	updateStatusTrash: function() {
+	updateStatusTrash = () => {
 		this.props.updatePostStatus( 'trash' );
-		recordEvent( 'Clicked Move to Trash' );
-	},
+		this.props.recordEvent( 'Clicked Move to Trash' );
+	}
 
-	updateStatusRestore: function() {
+	updateStatusRestore = () => {
 		this.props.updatePostStatus( 'restore' );
-		recordEvent( 'Clicked Restore' );
-	},
+		this.props.recordEvent( 'Clicked Restore' );
+	}
 
-	updateStatusDelete: function() {
+	updateStatusDelete = () => {
 		this.props.updatePostStatus( 'delete' );
-		recordEvent( 'Clicked Delete Page' );
-	},
-
-	copyPage: function() {
-		recordEvent( 'Clicked Copy Page' );
-	},
-} );
-
-export default connect(
-	( state, props ) => {
-		const site = getSite( state, props.page.site_ID );
-		const siteSlugOrId = get( site, 'slug' ) || get( site, 'ID', null );
-		const selectedSiteId = getSelectedSiteId( state );
-		const isPreviewable =
-			false !== isSitePreviewable( state, props.page.site_ID ) &&
-			site && site.ID === selectedSiteId;
-
-		return {
-			hasStaticFrontPage: hasStaticFrontPage( state, props.page.site_ID ),
-			isFrontPage: isFrontPage( state, props.page.site_ID, props.page.ID ),
-			isPostsPage: isPostsPage( state, props.page.site_ID, props.page.ID ),
-			isPreviewable,
-			previewURL: getPreviewURL( props.page ),
-			site,
-			siteSlugOrId,
-		};
-	},
-	{ setPreviewUrl, setLayoutFocus }
-)( updatePostStatus( localize( Page ) ) );
+		this.props.recordEvent( 'Clicked Delete Page' );
+	}
+
+	copyPage = () => {
+		this.props.recordEvent( 'Clicked Copy Page' );
+	}
+}
+
+const mapState = ( state, props ) => {
+	const site = getSite( state, props.page.site_ID );
+	const siteSlugOrId = get( site, 'slug' ) || get( site, 'ID', null );
+	const selectedSiteId = getSelectedSiteId( state );
+	const isPreviewable =
+		false !== isSitePreviewable( state, props.page.site_ID ) &&
+		site && site.ID === selectedSiteId;
+
+	return {
+		hasStaticFrontPage: hasStaticFrontPage( state, props.page.site_ID ),
+		isFrontPage: isFrontPage( state, props.page.site_ID, props.page.ID ),
+		isPostsPage: isPostsPage( state, props.page.site_ID, props.page.ID ),
+		isPreviewable,
+		previewURL: getPreviewURL( props.page ),
+		site,
+		siteSlugOrId,
+	};
+};
+
+const mapDispatch = {
+	setPreviewUrl,
+	setLayoutFocus,
+	recordEvent,
+	recordMoreOptions: partial( recordEvent, 'Clicked More Options Menu' ),
+	recordPageTitle: partial( recordEvent, 'Clicked Page Title' ),
+	recordEditPage: partial( recordEvent, 'Clicked Edit Page' ),
+	recordViewPage: partial( recordEvent, 'Clicked View Page' ),
+	recordStatsPage: partial( recordEvent, 'Clicked Stats Page' ),
+};
+
+export default flow(
```

Comment:
Well, having had some FP exposure in the past, I also defaulted to `compose`, but ever since lodash dropped that alias and instead converged on `flowRight` + `flow`, I took it as a sign to prefer the start-to-finish chaining of `flow`. Truth is I've come to like it. Think Elixir pipe operator:

```elixir
data
  |> map(f)
  |> filter(g)
```

It might be easier to read for newcomers too, perhaps. What do you think?